pkg/store/kubeconfig_store_rancher.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2021 The Kubeswitch authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+
+	"github.com/danielfoehrkn/kubeswitch/types"
+	"github.com/rancher/norman/clientbase"
+	managementClient "github.com/rancher/rancher/pkg/client/generated/management/v3"
+)
+
+func NewRancherStore(store types.KubeconfigStore) (*RancherStore, error) {
+	RancherStoreConfig := &types.StoreConfigRancher{}
+	if store.Config != nil {
+		buf, err := yaml.Marshal(store.Config)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		err = yaml.Unmarshal(buf, RancherStoreConfig)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal Rancher config: %w", err)
+		}
+	}
+
+	RancherAddress := RancherStoreConfig.RancherAddress
+
+	if len(RancherAddress) == 0 {
+		return nil, fmt.Errorf("when using the Rancher kubeconfig store, the address of Rancher has to be provided via SwitchConfig file")
+	}
+
+	RancherToken := RancherStoreConfig.RancherToken
+
+	if len(RancherToken) == 0 {
+		return nil, fmt.Errorf("when using the Rancher kubeconfig store, a Rancher API token must be provided via SwitchConfig file")
+	}
+
+	client, err := managementClient.NewClient(&clientbase.ClientOpts{
+		URL:      RancherAddress,
+		TokenKey: RancherToken,
+	})
```

Comment:
It might make sense to put the client instantiation into the StartSearch() function to execute it in a goroutine. Otherwise, it increases the perceived waiting time until the user sees the search screen (see the GKE store for an example).

Reply:
You would then also need to check in the `GetKubeconfigForPath()` if the client is already instantiated because if an index is used, `StartSearch()` is not executed.